packages/zoe/src/zoe.js

```diff
@@ -673,17 +928,24 @@ const makeZoe = (additionalEndowments = {}) => {
               proposal,
               currentAllocation: arrayToObj(amountsArray, userKeywords),
             };
+            // TODO split inviteHandle and offerHandle.
             // Since we have redeemed an invite, the inviteHandle is
             // also the offerHandle.
             offerTable.create(offerImmutableRecord, offerHandle);
             payoutMap.init(offerHandle, producePromise());
           };
 
           // Create result to be returned. Depends on `exit`
-          const makeRedemptionResult = _ => {
-            const redemptionResult = {
-              seat: handleToSeat.get(offerHandle),
+          const makeOfferResult = _ => {
+            const seatFn = handleToSeatFn.get(offerHandle);
+            // For now, the "remote" function call only works because
+            // the function is local. It cannot yet be remote
+            // in our system because functions are not yet passable.
+            const outcomeP = E(seatFn)(offerHandle);
+            const offerResult = {
+              offerHandle: HandledPromise.resolve(offerHandle),
```

Comment:
Why is the offerHandle a resolved promise?

Reply:
Good question! I missed this at first too. This has to do with the remaining payout liveness danger #776 . Say that I place an offer with moola(20), bucks(5). Say that the bucks issuer is malicious and never makes progress, but the moola issuer is well behaved. Say that the exit condition is onDemand. Such an offer never gets placed because the payments are never escrowed. After I get impatient enough I cancel. I should get my moola refund payout.

We're far from solving this whole thing, but we're making progress, and it is compat with our API (Hooray!). The key thing is to provide the offerResult record immediate, since that's where the cancelObj is found. Until cancelled, payout, outcome, and offerHandle all must remain unresolved promises, because the funds haven't yet been escrowed, the upcall hasn't been called, and there is no offer record to speak of.

Although I'm not making further progress towards solving it here, I made this a promise so I wouldn't make the situation worse.